src/Connection/Factory.php

```diff
@@ -167,10 +167,11 @@ protected function prepareConnection(NodeConnectionInterface $connection)
         if (isset($parameters->password) && strlen($parameters->password)) {
             $cmdAuthArgs = [$parameters->protocol, 'AUTH'];
 
-            isset($parameters->username) && strlen($parameters->username)
-                ? array_push($cmdAuthArgs, $parameters->username, $parameters->password)
-                : $cmdAuthArgs[] = $parameters->password;
+            if (!isset($parameters->username) || !strlen($parameters->username)) {
+                $parameters->username = 'default';
+            }
 
+            array_push($cmdAuthArgs, $parameters->username, $parameters->password);
             array_push($cmdAuthArgs, 'SETNAME', 'predis');
```

Comment:
Is it ok we're pushing a different command into the same cmd auth arguments?

Reply:
Unfortunately, we cannot use pipeline abstraction on connection level. Also the interface of connection accepts handshake commands one by one and add them into queue that will be flushed in connection time

`public function addConnectCommand(CommandInterface $command);`